xapian-core/include/xapian/weight.h

```diff
@@ -543,6 +548,7 @@ class XAPIAN_VISIBILITY_DEFAULT BM25Weight : public Weight {
 	    param_b = 0;
 	} else if (param_b > 1) {
 	    param_b = 1;
+	if (param_delta < 0) param_delta = 0;
```

Comment:
This is either not indented correctly, or is in the wrong place. I believe it should be after the closing parenthesis from the previous `if…else if`. (Unless `param_delta` should only be snapped to non-negative if `param_b` has been snapped to max-1, which seems unlikely.)